### ISSUE TITLE
"git hub pull-request" fails if pull request template exists, title never gets initialized

### ISSUE THREAD

**Issue by smemsh** [seveas/git-spindle#159]
I just had an ordinary master checkout, did a checkout -b to `feature/some-name`, made three commits (each had a standard log message i.e. title line, blank line, paragraph), and then ran `git hub pull-request`, getting this:
```
Traceback (most recent call last):
  File "/usr/local/bin/git-hub", line 11, in <module>
    GitHub().main()
  File "/usr/local/lib/python2.7/dist-packages/gitspindle/__init__.py", line 344, in main
    func(opts)
  File "/usr/local/lib/python2.7/dist-packages/gitspindle/github.py", line 1198, in pull_request
    title, body = self.edit_msg("%s\n\n%s" % (title,body), 'PULL_REQUEST_EDIT_MSG')
UnboundLocalError: local variable 'title' referenced before assignment
```

I looked at the code:
```
        if opts['--issue']:
            pull = parent.create_pull_from_issue(base=dst, head='%s:%s' % (repo.owner.login, src), issue=int(opts['--issue']))
            print("Pull request %d created %s" % (pull.number, pull.html_url))
            return

        body = self.find_template(repo, 'PULL_REQUEST_TEMPLATE')
        # 1 commit: title/body from commit
        if not body:
            if len(commits) == 1:
                title, body = self.gitm('log', '--pretty=%s\n%b', '%s^..%s' % (commits[0], commits[0])).stdout.split('\n', 1)
                title = title.strip()
                body = body.strip()
                accept_empty_body = not bool(body)

            # More commits: title from branchname (titlecased, s/-/ /g), body comments from shortlog
            else:
                title = src
                if '/' in title:
                    title = title[title.rfind('/') + 1:]
                title = title.title().replace('-', ' ')
                body = ""

        body += """
# Requesting a pull from %s/%s into %s/%s
#
# Please enter a message to accompany your pull request. Lines starting
# with '#' will be ignored, and an empty message aborts the request.
#""" % (repo.owner.login, src, parent.owner.login, dst)
        body += "\n# " + try_decode(self.gitm('shortlog', '%s/%s..%s' % (remote, dst, src)).stdout).strip().replace('\n', '\n# ')
        body += "\n#\n# " + try_decode(self.gitm('diff', '--stat', '%s^..%s' % (commits[0], commits[-1])).stdout).strip().replace('\n', '\n#')
        title, body = self.edit_msg("%s\n\n%s" % (title,body), 'PULL_REQUEST_EDIT_MSG')
        if not body and not accept_empty_body:
            err("No pull request message specified")
```

I don't get how this is happening; I don't have a template present, and I have three commits, so it should just hit this code:
```
            # More commits: title from branchname (titlecased, s/-/ /g), body comments from shortlog
            else:
                title = src
                if '/' in title:
                    title = title[title.rfind('/') + 1:]
                title = title.title().replace('-', ' ')
                body = ""
```

this code is passed through so it's unclear how title could not be set when it's used.

**Comment by smemsh** [comment]
so I found out there was indeed a `.github/pull_request_template.md` in my repo that was not know to me.  so the problem is, the code doesn't assign title anywhere, if the request template exists, so of course it won't work if referenced.

I think it will never work if any pull request template exists.  need some way to override and not use the template and / or somehow set title from template.

**Comment by igalic** [comment]
[hub](https://hub.github.com/) handles this by merging the template and the message: https://github.com/github/hub/blob/master/commands/pull_request.go#L374

**Comment by igalic** [comment]
I've created a pull request here: https://github.com/seveas/git-spindle/pull/160 but it needs a bit work

**Comment by seveas** [comment]
Fixed in 90cce34
